File: BackEnd/gamedex/src/main/java/cat/copernic/gamedex/entity/ResetPasswordRequest.java
package cat.copernic.gamedex.entity;

import org.springframework.data.mongodb.core.mapping.Document;
import lombok.Data;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

/**
 * Classe que representa una sol·licitud de restabliment de contrasenya.
 */
@Document(collection = "resetPasswordRequest")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class ResetPasswordRequest {

    /**
     * El nom d'usuari per a la sol·licitud de restabliment de contrasenya.
     */
    private String username;

    /**
     * El correu electrònic associat a la sol·licitud de restabliment de
     * contrasenya.
     */
    private String email;
}

File: FrontEnd/app/src/main/java/cat/copernic/grup4/gamedex/Core/Model/ResetPasswordRequest.kt
package cat.copernic.grup4.gamedex.Core.Model

/**
 * Classe que representa una sol·licitud de restabliment de contrasenya.
 *
 * @param username El nom d'usuari per a la sol·licitud de restabliment de contrasenya.
 * @param email El correu electrònic associat a la sol·licitud de restabliment de contrasenya.
 */
data class ResetPasswordRequest(
    val username: String,
    val email: String
)